.github/workflows/test.yaml

```diff
@@ -302,22 +303,27 @@ jobs:
   draft-publish-docs:
     name: Draft publish docs
     runs-on: ubuntu-22.04
+    needs: metadata-lint
     defaults:
       run:
         working-directory: ${{ inputs.working-directory }}
     steps:
       - uses: actions/checkout@v3
-      - name: Search for docs folder
-        id: docs-exist
-        run: echo "docs_exist=$([[ -d docs ]] && echo 'True' || echo 'False')" >> $GITHUB_OUTPUT
+      - name: Search for metadata and docs key in metadata
+        if: needs.metadata-lint.outputs.exists == 'true'
```

Comment:
Out of curiosity: is this condition required even with the "needs: metadata-lint"?

Reply:
Yes because `metadata-lint` will run even if there is no metadata file, thats why I added an output to that one to record whether the file exists and then this job reads that output